src/Particles.h

```diff
@@ -25,6 +25,13 @@ class Particles
     return _cell_particles;
   }
 
+  /// List of particles in a cell with (local) index \p cidx
+  const std::vector<int>& cell_particles(const int cidx) const
+  {
+    // TODO: assert validity of cidx
+    return _cell_particles[cidx];
+  }
+
```

Comment:
I'm not sure we need both of these calls (in the new minimalist world of dolfinx)... In C++ it is fine to take a reference to the `vector<vector<int>>&` and then just pull out the ones we need. For Python, maybe a bit less nice, but I think we can do that in the pybind layer, anyway.

Reply:
Removed the somewhat redundant specialization. Indeed a good point to aim at providing just one way of doing things